x/btcstaking/keeper/msg_server.go

```diff
@@ -622,16 +622,16 @@ func (ms msgServer) BTCUndelegate(goCtx context.Context, req *types.MsgBTCUndele
 		}
 		types.EmitEarlyUnbondedEvent(ctx, btcDel.MustGetStakingTxHash().String(), stakerSpendigTxHeader.Height)
 	case isStakeExpansion:
-		// stake expansion case: emit stake expansion activation event
-		delegatorUnbondingInfo = &types.DelegatorUnbondingInfo{
-			SpendStakeTx: req.StakeSpendingTx,
-		}
-		types.EmitStakeExpansionActivatedEvent(ctx,
+		// emit an early unbonded event for the delegation (the one that is being expanded)
+		types.EmitEarlyUnbondedEventByStakeExpansion(
+			ctx,
 			btcDel.MustGetStakingTxHash().String(),
 			spendStakeTxHash.String(),
-			req.StakeSpendingTxInclusionProof.Key.Hash.MarshalHex(),
-			req.StakeSpendingTxInclusionProof.Key.Index,
+			stakerSpendigTxHeader.Height,
```

Comment:
Should we add this field to `BTCStakingConsumerEvent_UnbondedDel` as well or it is not needed ?